packages/workflow/src/flags.ts

```diff
@@ -0,0 +1,30 @@
+export type SdkFlag = {
+  get id(): number;
+  get default(): boolean;
+};
+
+const flagsRegistry: Map<number, SdkFlag> = new Map();
+
+export const SdkFlags = {
+  /**
+   * This flag gates multiple fixes related to cancellation scopes and timers introduced in 1.10.2/1.10.3:
+   * - Cancellation of a non-cancellable scope no longer propagates to children scopes
+   *   (see https://github.com/temporalio/sdk-typescript/issues/1423).
+   * - CancellationScope.withTimeout(fn) now cancel the timer if `fn` completes before expiration
+   *   of the timeout, similar to how `condition(fn, timeout)` works.
+   * - Timers created using setTimeout can now be intercepted.
+   *
+   * @since Introduced in 1.10.2/1.10.3.
+   */
+  NonCancellableScopesAreShieldedFromPropagation: defineFlag(1, false),
+} as const;
+
+function defineFlag(id: number, def: boolean): SdkFlag {
+  const flag = { id, default: def };
```

Comment:
nit: I think the defaults will always be false but this is fine.

Reply:
nit: you might want to prevent duplicate registration for the same ID.

Reply:
> nit: I think the defaults will always be false but this is fine.

No. Assuming introduction of flags in two distinct releases (let say 1.10.2/1.10.3), the first release would default to false, and the second release would default to true.